### PR TITLE
ISSD-1406 Migrate market module AUI main.js file to purely javascript

### PR DESCRIPTION
**Pull request by Stephy6**
Steps to Verify
========
1. Go to Applications Menu -> Control Panel -> Marketplace.
2. Check that Store and Purchased portlets are rendered correctly and navigate through them.

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: modules/apps/marketplace/marketplace-store-web/src/main/resources/META-INF/resources/js/MarketplaceFrame.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/**
```
Comment (sandrodw3) [inline modules/apps/marketplace/marketplace-store-web/src/main/resources/META-INF/resources/js/MarketplaceFrame.js:1]: I think we can improve this a bit. 
- If it's just adapting the height to the content, I think we don't need to use the event, we can just do something like `frame.style.height = frame.contentWindow.document.body.scrollHeight + 'px';` so it adapts the height to its content
- Regarding the listener, note that you are passing different functions to add and to remove the listener. Take into account that `(event) => receiveMessage(event)` is not the same function as `receiveMessage`. You should declare a function that does exactly what you want and use it to add and remove the listener.  Please let me know if this is not clear and I'll explain you in detail 😇

Simplifying, I think we could leave something like:

```
export default function ({namespace: portletNamespace}) {
	const frame = document.getElementById(`${portletNamespace}frame`);

	const receiveMessage = () => {
		frame.style.height =
			frame.contentWindow.document.body.scrollHeight + 'px';
	};

	window.addEventListener('message', receiveMessage);

	return () => {
		window.removeEventListener('message', receiveMessage);
	};
}
```

Could you try it and test that everything works well?

**Review by sandrodw3 — COMMENTED** [review]
Hey @Stephy6! It seems it's working well! I just suggested a tiny refactor in the js component.

Also, I recommend you pay attention to the commit messages, the first commit has repeated text, and also instead of "Liferay-frontend" you should put "liferay-frontend:component", these small details improve the readability of the PR 😛.

File: modules/apps/marketplace/marketplace-store-web/build.gradle [hunk 2]
```diff
@@ -9,6 +9,8 @@ dependencies {
 	compileOnly group: "org.apache.felix", name: "org.apache.felix.http.servlet-api", version: "1.1.2"
 	compileOnly group: "org.osgi", name: "org.osgi.service.component.annotations", version: "1.4.0"
 	compileOnly project(":apps:application-list:application-list-api")
+	compileOnly project(":apps:frontend-taglib:frontend-taglib")
+	compileOnly project(":apps:frontend-taglib:frontend-taglib-clay")
```
Comment (sandrodw3) [inline modules/apps/marketplace/marketplace-store-web/build.gradle:13]: Is `frontend-taglib-clay` needed?

File: modules/apps/marketplace/marketplace-store-web/src/main/resources/META-INF/resources/js/MarketplaceFrame.js (hunk 1 above)
Comment (Stephy6) [reply modules/apps/marketplace/marketplace-store-web/src/main/resources/META-INF/resources/js/MarketplaceFrame.js:1]: Everything clear @sandrodw3 , thanks a lot!

File: modules/apps/marketplace/marketplace-store-web/build.gradle (hunk 2 above)
Comment (Stephy6) [reply modules/apps/marketplace/marketplace-store-web/build.gradle:13]: I don't think so

File: modules/apps/marketplace/marketplace-store-web/src/main/resources/META-INF/resources/js/MarketplaceFrame.js (hunk 1 above)
Comment (sandrodw3) [reply modules/apps/marketplace/marketplace-store-web/src/main/resources/META-INF/resources/js/MarketplaceFrame.js:1]: Yeah, I think so 😛

**Comment by sandrodw3** [comment]
ci:forward

**Review by sandrodw3 — APPROVED** [review]

**Comment by Tim-Cao** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/142795

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**